File: interop/java/DRandomUniform.java
package UniformImplementation;

import java.security.SecureRandom;
import java.math.BigInteger;

public class DRandomUniform {
  private static SecureRandom r = new SecureRandom();

  public static BigInteger Uniform(BigInteger n) {
    return BigInteger.valueOf(r.nextInt(n.intValue()));
  }
}

File: interop/go/DRandomUniform.go
package UniformImplementationxxx

import (
  "dafny"
	"math/rand"
)

var r = rand.New(rand.NewSource(99))

type FOO struct {}

var DRandomUniform = FOO{}

func (d FOO) Uniform(n dafny.Int) dafny.Int {
  return n
}





